### ISSUE TITLE
No devices found on Fedora KDE

### ISSUE THREAD

**Issue by cjbentley** [xournalpp/xournalpp#2273]
**Affects versions :**
 - OS: Fedora 32 KDE Spin
 - Desktop environment: KDE
 - Which version of libgtk do you use
```
gtk2.x86_64                                       2.24.32-7.fc32                      @fedora                
gtk3.x86_64                                       3.24.23-1.fc32                      @updates    
```
 - Version of Xournal++: 1.0.18
 - Installation method: Same issue with both Fedora repo and Flatpak install

**Describe the bug**
Xournal++ is unable to detect any devices on Fedora KDE. 
I do not believe this is a system configuration issue, as "regular" Xournal (instead of Xournal++) is able to detect the separate devices, and handles them properly.
I have tried Xournal++ with the new input system both enabled and disabled. I have also tried Xournal++ from both the Fedora repos, Flathub, and AppImage. The same behaviour persists. 

**To Reproduce**
1. Install Fedora KDE
2. Install Xournal++ from either Fedora repos or Flathub
3. Check Input System page in Xournal++ preferences.

**Expected behavior**
Touchscreen and Wacom digitizer should both show up, allowing user to designate one for scrolling, and one for writing.

**Screenshots of Problem**
Left side shows Xournal++, right side shows Xournal - devices recognized by Xournal but not Xournal++.
![xournal](https://user-images.githubusercontent.com/4556652/94087679-ee9c5f00-fdfd-11ea-898c-ba69b1ec5fcf.png)

**Additional context**
The following is a copy of dmesg output grep'd for "wacom" (digitizer used on this computer).
![dmesg](https://user-images.githubusercontent.com/4556652/94088023-cf520180-fdfe-11ea-9858-dc86641a79b8.png)

**Comment by EB2000** [comment]
I have exactly the same issue, also on Fedora 32 under KDE with a wacom device.

The device is not detected under xournal++ with either input method, while everything works fine with regular xournal, gimp or Write.

I have tried with fedora 32's package, and also with xournalpp-1.1.0-0.1.20200926git9417fca.fc32.x86_64
from COPR luya/xournalpp. 

When I run xournal++, I get the error:

* (com.github.xournalpp.xournalpp:9500): WARNING **: 21:43:15.505: No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!

and, in the "input system" tab of edit->preferences, it says:

No devices were found. This seems wrong - maybe file a bug report

My Wacom device, as it appears in dmesg:

input: Wacom Intuos Pro S Pen as /devices/pci0000:00/0000:00:14.0/usb1/1-7/1-7:1.0/bluetooth/hci0/hci0:1/0005:056A:0393.0004/input/input32
input: Wacom Intuos Pro S Finger as /devices/pci0000:00/0000:00:14.0/usb1/1-7/1-7:1.0/bluetooth/hci0/hci0:1/0005:056A:0393.0004/input/input33
input: Wacom Intuos Pro S Pad as /devices/pci0000:00/0000:00:14.0/usb1/1-7/1-7:1.0/bluetooth/hci0/hci0:1/0005:056A:0393.0004/input/input34

**Comment by voland66** [comment]
Could it be the same issue as the one here:
https://github.com/xournalpp/xournalpp/issues/1073

If it is the same issue, "unset GDK_CORE_DEVICE_EVENTS" should resolve the problem as described there.

**Comment by EB2000** [comment]
Yes, it must be the same issue: unsetting GDK_CORE_DEVICE_EVENTS solves the issue for me. Thank you!

**Comment by EB2000** [comment]
Ok, so the issue comes from the file /etc/xdg/plasma-workspace/env/gtk3_scrolling.sh, from kde-settings-plasma package. This was introduced to fix a scrolling bug 5 years ago, see https://bugzilla.redhat.com/show_bug.cgi?id=1226465

I'll put the same comment in the issue #1073, and then I'll stop commenting here.

**Comment by cjbentley** [comment]
I can confirm this fix works for me also.

At the linked bugzilla, the maintainers seem hesitant to reverse this workaround. (There are a few other people over there that ran into issues from this bug also.)

As regular Xournal doesn't have this issue, and it mostly seems to break a few specific applications, is the bug possibly on the Xournal++ end rather than Fedora's?

**Comment by mesonium** [comment]
> If it is the same issue, "unset GDK_CORE_DEVICE_EVENTS" should resolve the problem as described there.

I can confirm that this resolves my issue too!

---

I can confirm the issue on Fedora 35 KDE, both with 
* the repo build (1.1.0),
* the current offical flatpak version,
* as well as the build from git master ([ebaed95](https://github.com/xournalpp/xournalpp/commit/ebaed95f4784a4a5b6f276e6d3a9c3b339e60f31)) with libgtk 3.24.31, Kernel 5.16.5, libwacom-1.12.1-1, xorg-x11-drv-wacom-0.40.0-2feora

using a Wacom Intuos M tablet. The issue is that when hovering with the stylus, it's recognized as a mouse and thus also the configured pen buttons don't work but instead the configured mouse buttons are used, when using the pen's buttons and the pressure sensitivity is fixed.

Krita 4.4.5 works fine and does not show this issue on this system and with this tablet, so it looks very much like a Xournalpp bug.

```
> xsetwacom list devices
Wacom Intuos BT M Pad pad               id: 14  type: PAD       
Wacom Intuos BT M Pen stylus            id: 15  type: STYLUS
```

```
** (xournalpp:153706): WARNING **: 13:00:04.670: No device found. Is Xournal++ running in debugger / Eclipse...?
Probably this is the reason for not finding devices!
```
https://github.com/xournalpp/xournalpp/blob/f72fe7c85bf6ac0525350362b946e10df7a9371c/src/core/control/DeviceListHelper.cpp#L71
